src/debugger/server/Stepper.js

```diff
@@ -72,13 +72,32 @@ export class StepOverStepper extends Stepper {
   isComplete(ast: BabelNode, currentStackSize: number): boolean {
     if (!this.isAstLocationChanged(ast)) return false;
     if (currentStackSize <= this._startStackSize) {
-      // two cases here:
-      // if current stack length < starting stack length, the program must have
-      // hit an exception so this stepper is no longer relevant
-      // if current stack length === starting stack length, the program returned
+      // Two cases here:
+      // If current stack length < starting stack length, the program must have
+      // hit an exception so this stepper is no longer relevant.
+      // If current stack length === starting stack length, the program returned
       // to the same stack depth, so a step over is complete
       return true;
     }
     return false;
   }
 }
+
+export class StepOutStepper extends Stepper {
+  constructor(filePath: string, line: number, column: number, stackSize: number) {
+    super(filePath, line, column);
+    this._startStackSize = stackSize;
+  }
+  _startStackSize: number;
+
+  isComplete(ast: BabelNode, currentStackSize: number): boolean {
+    if (!this.isAstLocationChanged(ast)) return false;
```

Comment:
I do not think this is correct, source location may be not changing for recursive function call during step out. For example:
```
function foo() {
   if (depth > 10) {
     return;
   }
   foo();     <=== stop here.
}
```
In the above stop line, you issue a step-out, you may land at the same line in the caller frame.

Reply:
I see. I just tested it and it does affect correctness -- I had assumed it was simply an optimizing heuristic. Question: What benefit does this provide StepOver? It seems like its `if (currentStackSize <= this._startStackSize)` check would always be sufficient and correct.

Reply:
Upon further investigation, it seems like this same line causes the same bug in stepOver.

Reply:
Yes, we should fix for step-over case.